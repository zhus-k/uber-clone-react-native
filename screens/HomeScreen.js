import React from 'react';
import { StyleSheet, Text, View, SafeAreaView, Image } from 'react-native';
import tw from 'twrnc';
import NavOptions from '../components/NavOptions';
import { GooglePlacesAutocomplete } from 'react-native-google-places-autocomplete';
import { GOOGLE_MAPS_API_KEY } from '@env';
import { useDispatch } from 'react-redux';
import NavFavourites from '../components/NavFavourites';
import { setDestination, setOrigin } from '../slices/navSlice';

const HomeScreen = () => {
    const dispatch = useDispatch();

    return (
        <SafeAreaView style={tw`bg-white h-full`}>
            <View style={styles.container}>

                <Image
                    style={{
                        width: 100,
                        height: 100,
                        resizeMode: 'contain',
                    }}
                    source={{ uri: 'https://links.papareact.com/gzs' }}
                />

                <GooglePlacesAutocomplete
                    placeholder='From where?'
                    styles={{
                        container: {
                            flex: 0,
                        },
                        textInput: {
                            fontSize: 18,
                        },
                    }}
                    query={{
                        key: GOOGLE_MAPS_API_KEY,
                        language: 'en'
                    }}
                    onPress={(data, details = null) => {
                        dispatch(
                            setOrigin({
                                location: details.geometry.location,
                                description: data.description,
                            })
                        );
                        dispatch(
                            setDestination(null)
                        );
                    }}
                    listViewDisplayed='auto'
                    nearbyPlacesAPI='GooglePlacesSearch'
                    debounce={400}
                    enablePoweredByContainer={false}
                    minLength={2}
                    fetchDetails={true}
                    requestUrl={{
                        useOnPlatform: 'web', // or "all"
                        url:
                            'https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api', // or any proxy server that hits https://maps.googleapis.com/maps/api
                        headers: {
                            Authorization: `an auth token`, // if required for your proxy
                        },
                    }}
                />

                <NavOptions />

                <NavFavourites />

            </View>
        </SafeAreaView>
    );
};

export default HomeScreen;

const styles = StyleSheet.create({
    container: {
        flex: 0,
        width: '100%',
        padding: 10,
    },
});