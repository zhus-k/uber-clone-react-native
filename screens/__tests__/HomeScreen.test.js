import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react-native';

import configureStore from 'redux-mock-store';

import HomeScreen from '../HomeScreen';

jest.useFakeTimers();

const mockedDispatch = jest.fn();
const mockedNavigate = jest.fn();

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockedDispatch
}))

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigate
        }),
    }
})

describe('<HomeScreen />', () => {

    afterEach(cleanup);

    const mockStore = configureStore([]);

    test('renders correctly', () => {
        const store = mockStore({});

        const { getByTestId, getByText, queryByTestId, toJSON } = render(<HomeScreen />);

        const rendered = toJSON();

        expect(rendered).toMatchSnapshot();
    });
});