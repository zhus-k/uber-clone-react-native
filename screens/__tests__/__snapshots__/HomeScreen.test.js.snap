// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://links.papareact.com/gzs",
        }
      }
      style={
        Object {
          "height": 100,
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flex": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          clearButtonMode="while-editing"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="From where?"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 5,
                "flex": 1,
                "fontSize": 15,
                "height": 44,
                "marginBottom": 5,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              },
              Object {
                "fontSize": 18,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "id": "123",
            "image": "https://links.papareact.com/3pn",
            "screen": "MapScreen",
            "title": "Get a ride",
          },
          Object {
            "id": "456",
            "image": "https://links.papareact.com/28w",
            "screen": "EatsScreen",
            "title": "Order food",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#e5e7eb",
                "height": 300,
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
                "paddingBottom": 32,
                "paddingLeft": 24,
                "paddingRight": 8,
                "paddingTop": 16,
                "width": 160,
              }
            }
          >
            <View
              style={
                Object {
                  "opacity": 0.2,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://links.papareact.com/3pn",
                  }
                }
                style={
                  Object {
                    "height": 120,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                    "lineHeight": 28,
                    "marginTop": 8,
                  }
                }
              >
                Get a ride
              </Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#000",
                      "borderRadius": 16,
                      "marginTop": 16,
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#e5e7eb",
                "height": 300,
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
                "opacity": 1,
                "paddingBottom": 32,
                "paddingLeft": 24,
                "paddingRight": 8,
                "paddingTop": 16,
                "width": 160,
              }
            }
          >
            <View
              style={
                Object {
                  "opacity": 0.2,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://links.papareact.com/28w",
                  }
                }
                style={
                  Object {
                    "height": 120,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                    "lineHeight": 28,
                    "marginTop": 8,
                  }
                }
              >
                Order food
              </Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#000",
                      "borderRadius": 16,
                      "marginTop": 16,
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "destination": "Code Street, London, UK",
            "icon": "home",
            "id": "123",
            "location": "Home",
          },
          Object {
            "destination": "London Eye, London, UK",
            "icon": "briefcase",
            "id": "456",
            "location": "Work",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#d1d5db",
                    "borderRadius": 9999,
                    "marginRight": 16,
                    "paddingBottom": 12,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 12,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                    "lineHeight": 28,
                  }
                }
              >
                Home
              </Text>
              <Text
                style={
                  Object {
                    "color": "#6b7280",
                  }
                }
              >
                Code Street, London, UK
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e5e7eb",
                },
                Object {
                  "height": 0.5,
                },
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#d1d5db",
                    "borderRadius": 9999,
                    "marginRight": 16,
                    "paddingBottom": 12,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 12,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                    "lineHeight": 28,
                  }
                }
              >
                Work
              </Text>
              <Text
                style={
                  Object {
                    "color": "#6b7280",
                  }
                }
              >
                London Eye, London, UK
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
